### PR TITLE
Remove dependency on apps-common; update plugin versions

### PR DESCRIPTION
**Pull request by alechenninger**
Data management doesn't use anything in apps-common. At least for now. We already removed the overlay in #35 .

Also a few of the plugin versions were outdated.

### CODE REVIEW

**Comment by alechenninger** [comment]
@jewzaam Ready for review

File: data-mgmt-saml-auth/pom.xml [hunk 1]
```diff
@@ -73,14 +47,14 @@
...
-                        <version>2.1-alpha-3</version>
+                        <version>2.1.1</version>
                         <extensions>true</extensions>
                         <executions>
                             <execution>
                                 <goals>
                                     <goal>attached-rpm</goal>
                                 </goals>
-                                <phase>verify</phase>
+                                <phase>install</phase>
```
Comment (jewzaam) [inline data-mgmt-saml-auth/pom.xml]: Why was this changed?

File: data-mgmt-saml-auth/pom.xml [hunk 2]
```diff
@@ -48,19 +30,11 @@
...
                 <artifactId>maven-war-plugin</artifactId>
-                <version>2.4</version>
-                <configuration>
-                    <overlays>
-                        <overlay>
-                            <groupId>com.redhat.lightblue.applications</groupId>
-                            <artifactId>apps-common</artifactId>
-                        </overlay>
-                    </overlays>
-                </configuration>
```
Comment (jewzaam) [inline data-mgmt-saml-auth/pom.xml]: Why was the overlay removed?  Seems to be doing the same thing.  Is the plugin now treating any dependency with type `war` as an overlay?

Comment (jewzaam) [inline data-mgmt-saml-auth/pom.xml]: Ah, this is overlay of common.  Thought we did an overlay of the unauthenticated app originally.  Nevermind!
